### PR TITLE
EOS-25149: Aligned Chunk header implementation and btree integration.

### PR DESCRIPTION
**Pull request by NikhilBirgade**
Signed-off-by: Nikhil Kumar Birgade <nikhil.birgade@seagate.com>

# Problem Statement
- Aligned Chunk header implementation and btree integration.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -1574,11 +1593,20 @@ static int bnode_level(const struct nd *node)
 	return (node->n_type->nt_level(node));
 }
 
+#if 0
```
Comment (shashank-parulekar) [inline btree/btree.c:1596]: Remove this function if we do not need it.

File: btree/btree.c [hunk 2]
```diff
@@ -1878,11 +1906,12 @@ M0_INTERNAL void m0_btree_mod_fini(void)
 	m0_free(mod_get());
 }
 
+#if 0
```
Comment (shashank-parulekar) [inline btree/btree.c:1909]: Remove this one too.

File: btree/btree.c [hunk 3]
```diff
@@ -1897,6 +1926,7 @@ static bool addr_is_aligned(const void *addr)
 	return ((size_t)addr & ((1ULL << NODE_SHIFT_MIN) - 1)) == 0;
 }
 
+#if 0
```
Comment (shashank-parulekar) [inline btree/btree.c:1929]: Remove this one too.

File: btree/btree.c [hunk 4]
```diff
@@ -1921,12 +1951,20 @@ static bool segaddr_is_valid(const struct segaddr *seg_addr)
 static struct segaddr segaddr_build(const void *addr, int shift)
 {
 	struct segaddr sa;
+#if 0
```
Comment (shashank-parulekar) [inline btree/btree.c:1954]: Remove #if 0 code wherever present in this file.

File: btree/btree.c [hunk 5]
```diff
@@ -2292,7 +2339,8 @@ static int64_t bnode_get(struct node_op *op, struct td *tree,
 		node->n_seq           = m0_time_now();
 		node->n_ref           = 1;
 		node->n_txref         = 0;
-		node->n_size          = 1ULL << nt->nt_shift(node);
+		/* node->n_size          = 1ULL << nt->nt_shift(node); */
```
Comment (shashank-parulekar) [inline btree/btree.c:2342]: Remove commented code.

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: be/alloc.c [hunk 6]
```diff
@@ -419,6 +419,10 @@ static bool be_alloc_chunk_invariant(struct m0_be_allocator *a,
 
 	return _0C(c != NULL) &&
 	       _0C(be_alloc_chunk_is_in(a, ztype, c)) &&
+	       _0C(ergo(c->bac_chunk_align,
+			m0_addr_is_aligned((void *)c, c->bac_align_shift))) &&
+	       _0C(ergo(!c->bac_chunk_align,
+			m0_addr_is_aligned(&c->bac_mem, c->bac_align_shift))) &&
```
[the remark this answers is not in the document]
Comment (NikhilBirgade) [reply be/alloc.c:425]: Done.

File: btree/btree.c (hunk 1 above)
Comment (NikhilBirgade) [reply btree/btree.c:1596]: Done.

File: btree/btree.c (hunk 5 above)
Comment (NikhilBirgade) [reply btree/btree.c:2342]: Done.

File: btree/btree.c (hunk 3 above)
Comment (NikhilBirgade) [reply btree/btree.c:1929]: Done.

File: btree/btree.c (hunk 2 above)
Comment (NikhilBirgade) [reply btree/btree.c:1909]: Done.

File: btree/btree.c (hunk 4 above)
Comment (NikhilBirgade) [reply btree/btree.c:1954]: Done.

File: btree/btree.c [hunk 7]
```diff
@@ -901,7 +913,9 @@ enum {
 
 static struct segaddr  segaddr_build(const void *addr, int shift);
 static void           *segaddr_addr (const struct segaddr *addr);
+#if 0
 static int             segaddr_shift(const struct segaddr *addr);
+#endif
```
Comment (RadhaGulhane13) [inline btree/btree.c:918]: can we remove this code if not needed?

Comment (NikhilBirgade) [reply btree/btree.c:918]: Done.

**Review by RadhaGulhane13 — APPROVED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**